examples/keras_recipes/md/packaging_keras_models_for_wide_distribution.md

```diff
@@ -0,0 +1,516 @@
+`model = model_collection_xyz.AmazingModel()`
+
+* They are Keras functional models which means that they have a programmatically
+accessible graph of layers, for introspection or model surgery.
+
+This guide explains [how to use](#functional-subclassing-model) the Functional
+Subclassing pattern, and showcases its benefits for [programmatic model
+introspection](#model-introspection) and [model surgery](#model-surgery). It also shows
+two other best practices for sharable Keras models: [configuring
+models](#unconstrained-inputs) for the widest range of supported inputs, for example
+images of various sizes, and [using dictionary inputs](#model-with-dictionary-inputs) for
+clarity in more complex models.
+
+# Initialization
+
+The backend must be selected before keras 3 is first imported.
+
+
+```python
+# backend selection in Keras 3. Supported values are jax, torch and tensorflow
+import os
+
+os.environ["KERAS_BACKEND"] = "tensorflow"
+
+import keras
+
+print("Keras version", keras.version())
+print("Keras is running on", keras.config.backend())
+```
+
+<div class="k-default-codeblock">
+```
+Keras version 3.0.1
+Keras is running on tensorflow
+
+```
+</div>
+# Dataset
+
+Let's load an MNIST dataset so that we have something to train with.
+
+
+```python
+import tensorflow as tf  # only for tf.data
+
+# tf.data is a great API for putting together a data stream.
+# It works wether you use the TensorFlow, PyTorch or Jax backend,
+# as long as you use it in the data stream only and not inside of a model.
+
+BATCH_SIZE = 256
+
+(x_train, train_labels), (x_test, test_labels) = tf.keras.datasets.mnist.load_data()
+
+train_data = tf.data.Dataset.from_tensor_slices((x_train, train_labels))
+train_data = train_data.map(
+    lambda x, y: (tf.expand_dims(x, axis=-1), y)
+)  # 1-channel monochrome
+train_data = train_data.batch(BATCH_SIZE)
+train_data = train_data.cache()
+train_data = train_data.shuffle(5000, reshuffle_each_iteration=True)
+train_data = train_data.repeat()
+
+test_data = tf.data.Dataset.from_tensor_slices((x_test, test_labels))
+test_data = test_data.map(
+    lambda x, y: (tf.expand_dims(x, axis=-1), y)
+)  # 1-channel monochrome
+test_data = test_data.batch(10000)
+test_data = test_data.cache()
+
+STEPS_PER_EPOCH = len(train_labels) // BATCH_SIZE
+EPOCHS = 5
+```
+
+# Functional Subclassing Model
+
+The model is wrapped in a class so that end users can instantiate it normally by calling
+the constructor `MnistModel()` rather than calling a factory function. The factory
+function `make_functional_model()` exists
+but users don't need to care about it. It is called in the constructor.
+
+
+```python
+
+class MnistModel(keras.Model):
+    def __init__(self, **kwargs):
+        inputs, outputs = self.make_functional_model()
+        # A Keras Functional model is created by calling keras.Model(inputs, outputs)
+        super().__init__(inputs=inputs, outputs=outputs, **kwargs)
+
+    @staticmethod
+    def make_functional_model():
```

Comment:
Prefer inlining this content in the constructor (it is the entire constructor's contents, so splitting it out does not offer complexity reduction benefits)

Reply:
fixed